diff -Naur perl-5.12.3.orig/mg.c perl-5.12.3/mg.c
--- perl-5.12.3.orig/mg.c	2011-01-09 21:20:57.000000000 +0100
+++ perl-5.12.3/mg.c	2011-04-05 12:11:37.837035780 +0200
@@ -2898,7 +2898,7 @@
     PUSHSTACKi(PERLSI_SIGNAL);
     PUSHMARK(SP);
     PUSHs(sv);
-#if defined(HAS_SIGACTION) && defined(SA_SIGINFO)
+#if 0 && defined(HAS_SIGACTION) && defined(SA_SIGINFO)
     {
 	 struct sigaction oact;
 
